node/events-to-ga/src/utils.js

```diff
@@ -0,0 +1,59 @@
+/**
+ * Throws an error if any of the keys are missing from the object
+ * @param {*} obj
+ * @param {string[]} keys
+ * @throws {Error}
+ */
+export function throwIfMissing(obj, keys) {
+  const missing = [];
+  for (let key of keys) {
+    if (!(key in obj) || !obj[key]) {
+      missing.push(key);
+    }
+  }
+  if (missing.length > 0) {
+    throw new Error(`Missing required fields: ${missing.join(", ")}`);
+  }
+}
+
+/**
+ * returns whether the request headers and appropriate message are ok
+ * @param {*} req
+ * @returns {{message:string,containsReqHeaders:boolean}}
+ */
+export function verifyHeaders(req) {
+  let containsReqHeaders = true;
+  let message = "";
+  if (req.headers["x-appwrite-user-id"] == "") {
+    message = `x-appwrite-user-id value in req.headers is not there`;
+    containsReqHeaders = false;
+  } else if (req.headers["x-appwrite-trigger"] != "event") {
+    message = `Not triggered by event but by ${req.headers["x-appwrite-trigger"]}`;
+    containsReqHeaders = false;
+  } else if (req.headers["x-appwrite-event"] == "") {
+    message = `x-appwrite-event value in req.headers is not there`;
+    containsReqHeaders = false;
+  }

```

Comment:
using if elses like this can be a little confusing. It might be clearer to just use separate if statements and add a throw inside each if to exit early (which is also a good pattern).